nengo/builder/optimizer.py

```diff
@@ -211,7 +211,8 @@ def _perform_merges(self, only_merge_ops_with_view):
             op_replacements.update(opr)
             sig_replacements.update(sigr)
             del by_type[tp]  # no need to process tp again in the second loop
-            if not only_merge_ops_with_view and len(opr) > 0:
+            n_changes = sum(new is not old for old, new in iteritems(opr))
```

Comment:
We don't need the exact number, do we? An `any` instead of `sum` might be more efficient.